backend/app/core/browse.py

```diff
@@ -1,12 +1,104 @@
 """Functions to support browsing the RDS document structure"""
 
+from time import perf_counter
+from typing import List, Optional
+from pydantic import BaseModel
+from app.db.models.document import Category, Document, Event, Geography, DocumentType
+from app.api.api_v1.schemas.metadata import Event as MetaEvent
 from app.api.api_v1.schemas.search import (
-    SearchRequestBody,
+    SearchResponseDocument,
     SearchResponseBody,
 )
+from sqlalchemy import extract
+from sqlalchemy.orm import Session
 
 
-def browse_rds(request: SearchRequestBody) -> SearchResponseBody:
+class BrowseArgs(BaseModel):
+    """Arguements for the browse_rds function"""
+
+    geography_id: Optional[int] = None
+    country_code: Optional[str] = None
+    start_year: Optional[int] = None
+    end_year: Optional[int] = None
+    category: Optional[str] = None
+
+
+def to_search_resp_doc(row: dict) -> SearchResponseDocument:
+    return SearchResponseDocument(
+        document_id=row["id"],
+        document_name=row["name"],
+        document_description=row["description"],
+        document_date=str(row["publication_ts"]),
+        document_category=row["category"],
+        document_country_code=row["country_code"],
+        document_country_english_shortname=row["country_name"],
+        # Stuff we don't use
+        document_source_name="",
+        document_type="",
+        document_source_url="",
+        document_url="",
+        document_content_type="",
+        document_title_match=False,
+        document_description_match=False,
+        document_passage_matches=[],
```

Comment:
this looks so much like a `DocumentOverviewResponse`... perhaps we should be trying to unify them...?

Reply:
Let's chat - as this is for a response the same as search (as browse needs to return the same structure as search I think) .... /cc @PatFawbertMills

Reply:
Talked - this will be coming with BAK-1151